custom/aaa/models.py

```diff
@@ -1212,3 +1207,70 @@ def _dictfetchone(cursor):
         return ret[0]
 
     return {}
+
+
+class DenormalizedAWC(models.Model):
+    domain = models.TextField()
+    state_id = models.TextField()
+    district_id = models.TextField()
+    block_id = models.TextField()
+    supervisor_id = models.TextField()
+    awc_id = models.TextField(unique=True)
+
+    @classmethod
+    def build(cls, domain):
+        cls.objects.filter(domain=domain).delete()
+        domain_locations = SQLLocation.objects.filter(domain=domain).values(
```

Comment:
loading all might not work for ICDS (~2M locations at scale) but should be fine for REACH. Its quite light since I only load the 4 necessary values. even at scale for ICDS i don't think it would take more than 400 MB which is probably fine